includes/class-dispatcher.php

```
@@ -64,10 +64,14 @@ public static function process_outbox( $id ) {
 		$activity = new Activity();
 		$activity->set_type( $type );
 		$activity->set_id( $outbox_item->guid );
-		// Pre-fill the Activity with data (for example cc and to).
-		$activity->set_object( Activity::init_from_json( $outbox_item->post_content ) );
 		$activity->set_actor( Actors::get_by_id( $outbox_item->post_author )->get_id() );
 
+		// Pre-fill the Activity with data (for example cc and to).
+		$activity_object = Activity::init_from_json( $outbox_item->post_content );
+		if ( ! is_wp_error( $activity_object ) ) {
+			$activity->set_object( $activity_object );
```

Should't we fail here instead of working around it and nulling the object?

---

What would failing look like in this case? Publish the outbox item so it doesn't get retried and then return?